applications/stepswitch/src/stepswitch_bridge.erl

```diff
@@ -693,6 +695,113 @@ send_deny_emergency_response(OffnetReq, ControlQ) ->
                                       ),
     kz_call_response:send(CallId, ControlQ, Code, Cause, Media).
 
+-spec send_emergency_bridge_notification(state()) -> 'ok'.
+send_emergency_bridge_notification(#state{resource_req=OffnetReq}=State) ->
+    Setters = [{fun set_emergency_call_meta/2, State}
+              ,{fun set_emergency_device_meta/2, OffnetReq}
+              ,{fun set_emergency_address_meta/2, OffnetReq}
+              ,{fun set_emergency_user_meta/2, OffnetReq}
+              ,{fun set_default_headers/2, OffnetReq}
+              ],
+
+    Props = lists:foldl(fun({F, O}, A) -> F(O, A) end, [], Setters),
+    kapps_notify_publisher:cast(Props, fun kapi_notifications:publish_emergency_bridge/1).
+
+-spec set_default_headers(kapi_offnet_resource:req(), kz_term:proplist()) -> kz_term:proplist().
+set_default_headers(_OffnetReq, Props) ->
+    Props ++ kz_api:default_headers(?APP_NAME, ?APP_VERSION).
+
+-spec set_emergency_call_meta(state(), kz_term:proplist()) -> kz_term:proplist().
+set_emergency_call_meta(#state{resource_req=OffnetReq}=State, Props) ->
+    [{<<"Call-ID">>, kapi_offnet_resource:call_id(OffnetReq)}
+    ,{<<"Account-ID">>, kapi_offnet_resource:account_id(OffnetReq)}
+    ,{?KEY_E_CALLER_ID_NUMBER, kapi_offnet_resource:emergency_caller_id_number(OffnetReq)}
+    ,{?KEY_E_CALLER_ID_NAME, kapi_offnet_resource:emergency_caller_id_name(OffnetReq)}
+    ,{?KEY_OUTBOUND_CALLER_ID_NUMBER, kapi_offnet_resource:outbound_caller_id_number(OffnetReq)}
+    ,{?KEY_OUTBOUND_CALLER_ID_NAME, kapi_offnet_resource:outbound_caller_id_name(OffnetReq)}
+    ,{<<"Emergency-Test-Call">>, maybe_emergency_test_call(State)}
+    ,{<<"Emergency-To-DID">>, kapi_offnet_resource:to_did(OffnetReq)}
+     | Props
+    ].
+
+
+-spec set_emergency_device_meta(kapi_offnet_resource:req(), kz_term:proplist()) -> kz_term:proplist().
+set_emergency_device_meta(OffnetReq, Props) ->
+    AccountID = kapi_offnet_resource:account_id(OffnetReq),
+    DeviceID = kz_json:get_ne_value(<<"Authorizing-ID">>, kapi_offnet_resource:requestor_custom_channel_vars(OffnetReq), 'undefined'),
+    OwnerID =kz_json:get_ne_value(<<"Owner-ID">>, kapi_offnet_resource:requestor_custom_channel_vars(OffnetReq), 'undefined'),
+    {'ok', DeviceJObj} = kzd_devices:fetch(AccountID, DeviceID),
+    [{<<"Authorizing-ID">>, DeviceID}
+    ,{<<"Owner-ID">>, OwnerID}
```

Comment:
What should happen if no "owner_id" is set but a user is hotdesked in here?

Reply:
Per discussion a device with multiple users hotdesked in will behave like a device without an owner, as there is now way to differentiate between the hotdesked users.